tensorboard/plugins/histogram/summary_v2.py

```diff
@@ -196,53 +196,50 @@ def _buckets(data, bucket_count=None):
-        is_empty = tf.equal(tf.size(input=data), 0)
-
-        def when_empty():
+        data_size = tf.size(input=data)
+        if tf.equal(data_size, 0):
             return tf.constant([], shape=(0, 3), dtype=tf.float64)
 
-        def when_nonempty():
-            min_ = tf.reduce_min(input_tensor=data)
-            max_ = tf.reduce_max(input_tensor=data)
-            range_ = max_ - min_
-            is_singular = tf.equal(range_, 0)
-
-            def when_nonsingular():
-                bucket_width = range_ / tf.cast(bucket_count, tf.float64)
-                offsets = data - min_
-                bucket_indices = tf.cast(
-                    tf.floor(offsets / bucket_width), dtype=tf.int32
-                )
-                clamped_indices = tf.minimum(bucket_indices, bucket_count - 1)
-                one_hots = tf.one_hot(clamped_indices, depth=bucket_count)
-                bucket_counts = tf.cast(
-                    tf.reduce_sum(input_tensor=one_hots, axis=0),
-                    dtype=tf.float64,
-                )
-                edges = tf.linspace(min_, max_, bucket_count + 1)
-                # Ensure edges[-1] == max_, which TF's linspace implementation does not
-                # do, leaving it subject to the whim of floating point rounding error.
-                edges = tf.concat([edges[:-1], [max_]], 0)
-                left_edges = edges[:-1]
-                right_edges = edges[1:]
-                return tf.transpose(
-                    a=tf.stack([left_edges, right_edges, bucket_counts])
-                )
-
-            def when_singular():
-                center = min_
-                bucket_starts = tf.stack([center - 0.5])
-                bucket_ends = tf.stack([center + 0.5])
-                bucket_counts = tf.stack(
-                    [tf.cast(tf.size(input=data), tf.float64)]
-                )
-                return tf.transpose(
-                    a=tf.stack([bucket_starts, bucket_ends, bucket_counts])
-                )
+        min_ = tf.reduce_min(input_tensor=data)
+        max_ = tf.reduce_max(input_tensor=data)
+        range_ = max_ - min_
+        is_singular = tf.equal(range_, 0)
+
+        bucket_width = range_ / tf.cast(bucket_count, tf.float64)
+        offsets = data - min_
+        # For singular data, indices should simply be
+        # [bucket_count - 1] * data_size, since only the last bucket (closed
+        # with a format of [v, v]) will have nonzero count.
+        singular_bucket_indices = tf.fill([data_size], bucket_count - 1)
+        non_singular_bucket_indices = tf.cast(
+            tf.floor(offsets / bucket_width), dtype=tf.int32
+        )
+        bucket_indices = (
+            singular_bucket_indices
+            if is_singular
+            else non_singular_bucket_indices
+        )
+        clamped_indices = tf.minimum(bucket_indices, bucket_count - 1)
+        # Cast to float64 to prevent overflow beyond limt 2^24 in reduce_sum below.
```

Comment:
On further thought (which I should have suggested earlier), I think it's probably best if we break this fix out separately and do it first, since it's a small fix to the existing code, rather than mingling it with the v3 format changes.  Do you mind taking that on?  It'd involve just the approximately-one-line change here, plus adding a test to confirm it stays fixed.